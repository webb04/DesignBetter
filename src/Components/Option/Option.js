import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
    border: none;
    background: none;
    font-family: "Maison Neue";
    width: 100%;
    letter-spacing: 0.2px;
    transition: transform 0.2s ease-in-out;

    &:hover {
        transform: translate(0px, -5px);
        cursor: pointer;
    }

    &:active, &:focus {
        outline: 0;
        color: inherit;
    }
`;

const Title = styled.h3`
    margin-bottom: 5px;
`

const Subtitle = styled.h4`
    color: #8A959E;
    font-weight: 300;
    margin-top: 5px;
`

const Progress = styled.div`
    width: 100%;
    background: rgb(222, 224, 227);
    height: 1px;
    border-radius: 3px;
`

const Bar = styled.div.attrs({
    style: ({ selected, progress }) => ({
        display: selected ? 'block' : 'none',
        width: `${progress}%`,
    })
})`
    background: #FF3366;
    height: 1px;
    border-radius: 3px;
`

const Option = ({ onClick, title, subtitle, progress, selected }) => (
    <StyledButton onClick = {onClick}>
        <Title>{title}</Title>
        <Subtitle>{subtitle}</Subtitle>
        <Progress>
            <Bar className="bar" progress={progress} selected={selected}/>
        </Progress>
    </StyledButton>
)

Option.propTypes = {
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    progress: PropTypes.number.isRequired,
    selected: PropTypes.bool.isRequired,
}

export default Option;