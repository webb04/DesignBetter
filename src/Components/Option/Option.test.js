import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Option from './Option';
import { shallow } from 'enzyme';

const selected = {
    onClick: () => {},
    title: 'Built-in animations',
    subtitle: 'Show user interactions',
    progress: 50,
    selected: true,
};

const unselected = {
    onClick: () => {},
    title: 'Built-in animations',
    subtitle: 'Show user interactions',
    progress: 10,
    selected: false,
};

it('Shows correct progress', () => {
    const selectedOption = renderer.create(<Option {...selected}/>).toJSON();
    expect(selectedOption.children[2].children[0].props.style.width).toEqual('50%');
    expect(selectedOption.children[2].children[0].props.style.display).toEqual('block');

    const unselectedOption = renderer.create(<Option {...unselected}/>).toJSON();
    expect(unselectedOption.children[2].children[0].props.style.width).toEqual('10%');
    expect(unselectedOption.children[2].children[0].props.style.display).toEqual('none');
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Option {...selected}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});