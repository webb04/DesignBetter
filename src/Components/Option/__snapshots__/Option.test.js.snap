// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bdVaJa bIgjLa"
  onClick={[Function]}
>
  <h3
    className="sc-bwzfXH kXBrIV"
  >
    Built-in animations
  </h3>
  <h4
    className="sc-htpNat dpENTR"
  >
    Show user interactions
  </h4>
  <div
    className="sc-bxivhb kDOLAr"
  >
    <div
      className="bar sc-ifAKCX dxjrqa"
      selected={true}
      style={
        Object {
          "display": "block",
          "width": "50%",
        }
      }
    />
  </div>
</button>
`;
